pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -208,11 +212,25 @@ func (a *Bootstrap) getTemplateData(installConfig *types.InstallConfig, proxy *c
 		logrus.Debugf("Using internal constant for release image %s", releaseImage)
 	}
 
+	var deployImage string
+	// baremetal requires some additional templateData to download the RHCOS image
+	// so we can deploy the masters with the correct image from rhcos.json
+	if installConfig.Platform.Name() == bmtypes.Name {
+		var err error
+		deployImage, err = rhcos.Baremetal()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		deployImage = releaseImage
```

Comment:
I'd prefer to pass an empty string than confuse the meaning of deployImage - i.e. if it is set, it should be an rhcos image URL, not a release image pullspec